integration-tests/src/integration-tests/java/oracle/kubernetes/operator/utils/TestUtils.java

```diff
@@ -0,0 +1,544 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.logging.Logger;
+import java.security.KeyStore;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.glassfish.jersey.jsonp.JsonProcessingFeature;
+
+public class TestUtils {
+  private static final Logger logger = Logger.getLogger("OperatorIT", "OperatorIT");
+
+  public static final int MAX_ITERATIONS_POD = 50; //50 * 5 = 250 seconds
+  public static final int WAIT_TIME_POD = 5;
+
+  public static String executeCommand(String command) {
+    StringBuffer output = new StringBuffer();
+    Process p;
+    try {
+      p = Runtime.getRuntime().exec(command);
+      p.waitFor();
+      BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+      BufferedReader errReader = new BufferedReader(new InputStreamReader(p.getErrorStream()));
+
+      //in some cases u may want to read process error stream as well
+      String line = "";
+      while ((line = reader.readLine()) != null) {
+        output.append(line + "\n");
+      }
+
+      while ((line = errReader.readLine()) != null) {
+        output.append(line + "\n");
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return output.toString();
+  }
+
+  public static String executeCommand(String commandArgs[]) {
+    StringBuffer output = new StringBuffer();
+    Process p;
+    try {
+      p = Runtime.getRuntime().exec(commandArgs);
+      p.waitFor();
+      BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+      BufferedReader errReader = new BufferedReader(new InputStreamReader(p.getErrorStream()));
+
+      //in some cases u may want to read process error stream as well
+      String line = "";
+      while ((line = reader.readLine()) != null) {
+        output.append(line + "\n");
+      }
+
+      while ((line = errReader.readLine()) != null) {
+        output.append(line + "\n");
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return output.toString();
+  }
+
+  public static boolean executeCommand(String command, String resultString) {
+    String output = executeCommand(command);
+    if (output.contains(resultString)) return true;
+    else return false;
+  }
+  /** @param cmd - kubectl get pod <podname> -n namespace */
+  public static void checkPodReady(String podName, String domainNS) {
+
+    int i = 0;
+    StringBuffer cmd = new StringBuffer();
+    cmd.append("kubectl get pod ").append(podName).append(" -n ").append(domainNS);
+
+    while (i < MAX_ITERATIONS_POD) {
```

Comment:
How about adding a utility that does the looping where you can pass in the code that does the check?
v.s. repeating the while loop, sleep, max iterations stuff a bunch of places.